sdk/ml/azure-ai-ml/azure/ai/ml/entities/_autogen_entities/models/_patch.py

```diff
@@ -0,0 +1,174 @@
+        return _get_rest_field_type(rest_field._type.args[0])
+    if hasattr(rest_field, "func") and func_to_attr_type.get(rest_field.func.__name__):
+        return func_to_attr_type[rest_field.func.__name__]
+    if hasattr(rest_field, "args"):
+        return _get_rest_field_type(rest_field.args[0])
+    return rest_field
+
+
+class ValidationMixin:
+    def _validate(self) -> None:
+        # verify types
+        for attr, rest_field in self._attr_to_rest_field.items():
+            try:
+                attr_value = self.__getitem__(attr)
+                attr_type = type(attr_value)
+            except KeyError:
+                if rest_field._visibility and "read" in rest_field._visibility:
+                    # read-only field, no need to validate
+                    continue
+                if rest_field._type.func.__name__ != "_deserialize_with_optional":
+                    # i'm required
+                    raise ValueError(f"attr {attr} is a required property for {self.__class__.__name__}")
+            else:
+                if getattr(attr_value, "_is_model", False):
+                    return attr_value._validate()
+                rest_field_type = _get_rest_field_type(rest_field)
+                if attr_type != rest_field_type:
+                    raise ValueError(f"Type of attr {attr} is of type {attr_type}, not {rest_field_type}")
+
+
+@experimental
+class MarketplacePlan(_MarketplacePlan):
+    pass
+
+
+@experimental
+class ServerlessEndpoint(_ServerlessEndpoint, ValidationMixin):
+
+    system_data: Optional[SystemData] = rest_field(visibility=["read"])
+    """System data of the endpoint."""
+
+    def _to_rest_object(self) -> RestServerlessEndpoint:
+        return RestServerlessEndpoint(
+            properties=RestServerlessEndpointProperties(
+                model_settings=RestModelSettings(model_id=self.model_id),
+            ),
+            tags=self.tags,
+            sku=RestSku(name="Consumption"),
+            location=self.location,
+        )
+
+    @classmethod
+    def _from_rest_object(cls, obj: RestServerlessEndpoint) -> "ServerlessEndpoint":
+        return cls(
+            name=obj.name,
+            id=obj.id,
+            tags=obj.tags,
+            location=obj.location,
+            auth_mode=obj.properties.auth_mode,
+            provisioning_state=camel_to_snake(obj.properties.provisioning_state),
+            model_id=obj.properties.model_settings.model_id,
+            scoring_uri=obj.properties.inference_endpoint.uri,
+            system_data=SystemData._from_rest_object(obj.system_data),
+            headers=obj.properties.inference_endpoint.headers,
+        )
+
+    def as_dict(self, *, exclude_readonly: bool = False) -> Dict[str, Any]:
+        d = super().as_dict(exclude_readonly=exclude_readonly)
+        d["system_data"] = json.loads(json.dumps(self.system_data._to_dict()))
+        return d
+
+
+ServerlessEndpoint.__doc__ += _ServerlessEndpoint.__doc__.strip() + """
```

Comment:
This is very smart! Thank you. Do we need to add doc string for other top level properties like `model_id` and `scoring_uri`?